src/Engine/Rendering/Renderer.cpp

```diff
@@ -720,6 +722,11 @@ int Renderer::buildAllRenderTechniques() const {
     return 0;
 }
 
+Scalar Renderer::getDepth( int, int ) {
+    LOG( logERROR ) << "call to not implemented Renderer::getDepth (return -1) , please override";
+    return -1_ra;
```

Comment:
This does not match the documentation of the function that says return value is in [0, 1].
As this function must never be called we have four possibilities : 
 - make it pure virtual (force to implement it in concrete renderers)
 - make it return a "valid value", either 1 to rotate around the background, either 0.5 to rotate around the center of the scene, or 0 to rotate around the near plane.
 - Add a note in the documentation saying that the return value is {-1} U [0, 1] with the value -1 indicating that the service is not implemented and that the value should not be used.
 - We discussed to augment the picking result with the picked depth and to use it in Viewer instead of getDepth. I think it might be a better approach to do that but is it in the scope of this PR ?

Reply:
fine with picking version, it solves lot's of question. 
I'll test it now.

Reply:
Some sort of resolved, but preparePicking is still in a not really finished state.